src/invidious/jsonfilter.cr

```diff
@@ -0,0 +1,251 @@
+require "json"
+
+module JSONFilter
+  alias BracketIndex = Hash(Int64, Int64)
+
+  alias GroupedFieldsValue = String | Array(String) | Array(GroupedFieldsValue)
```

Comment:
You can simplify this to:
```crystal
alias GroupedFieldsValue = String | Array(GroupedFieldsValue)
```

Since the inner value in `Array(GroupedFieldsValue)` can be a string.